src/Chainweb/Pact/PactService.hs

```diff
@@ -1293,14 +1326,21 @@ execPreInsertCheckReq txs = do
     b <- liftIO $ _cpGetLatestBlock cp
     case b of
         Nothing -> return $! V.map Right txs
-        Just (h, ha) ->
-            withCheckpointer (Just (h+1, ha)) "execPreInsertCheckReq" $ \pdb -> do
-                now <- liftIO getCurrentTimeIntegral
+        Just (parentHeight, parentHash) -> do
```

Comment:
I don't understand why the pre-insert check needs to change? Current time would seem to make sense here.

Reply:
The old check was just an approximation, because the true reference time for tx timing validation wasn't yet known. With the new behavior we can make the check precise.

Once the transition is complete, the code paths for the old check can be removed.